src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -162,17 +311,60 @@ public void handleEvent(RegistryEvent.Register<?> event) {
         return entriesView;
     }
 
+    /**
+     * @return The registry name stored in this deferred register. Useful for creating new deferred registers based on an existing one.
+     */
+    @Nullable
+    public ResourceLocation getRegistryName()
+    {
+        return this.registryName;
+    }
+
+    private Supplier<IForgeRegistry<T>> makeRegistry(final ResourceLocation registryName, final Class<T> superType, final Supplier<RegistryBuilder<T>> sup) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    private Supplier<IForgeRegistry<T>> makeRegistry(final ResourceLocation registryName, final Class<T> superType, final Supplier<RegistryBuilder<T>> sup)
    {
```